📖 add documentation about namingStrategy in ClusterClasses (#9433)
kubernetes-sigs/cluster-api: docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

</aside>

## ClusterClass with custom naming strategies

The controller needs to generate names for new objects when a Cluster is getting created

> from a ClusterClass. These names have to be unique on a per namespace base which mostly is
> accomplished by concatenating the cluster name in some way with a random suffix.

Reviewer: ```suggestion
from a ClusterClass. These names have to be unique for each namespace. The naming strategy enables this  by concatenating the cluster name with a random suffix.
```
